pkg/operator/ceph/config/keyring/store.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package keyring provides methods for accessing keyrings for Ceph daemons stored securely in
+// Kubernetes.
+package keyring
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "op-cfg-keyring")
+
+const (
+	keyringStorePath = "/etc/ceph/keyring-store/"
+	keyringFileName  = "keyring"
+)
+
+// SecureStore is a helper to store Ceph daemon keyrings as Kubernetes secrets.
+type SecureStore struct {
```

Comment:
Is there any need for this to use secrets? All the settings would be in the crd in plain text, right? Keeping them readable in a configmap seems like a helpful thing for debugging.

Reply:
I thought it best to keep them in secrets because they contain the keys daemons use to auth with the mons. I was under the impression that this would be the best practice because we chose to specifically remove the mon/admin keys from any container logs. My gut thinks the extra security is probably worth less readability, but I am far from a security expert in kubernetes. A ConfigMap may be secure enough, but I'd like to have some sort of concrete reasoning about that before I make a potentially unsecure decision.